### PR TITLE
Update DG: Appendix

### PR DESCRIPTION
**Pull request by FredericChow00**
## What
Edited the Requirements Section of Appendex

## Why
To allow developers to understand our product better

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -257,32 +257,48 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *` | Busy Hiring Manager | search for applicants  | I can view details of specific applicants |
-| `* * ` | Busy Hiring Manager | view the dates of interviews for all shortlisted applicants | I can better plan my future working days |
-| `* * ` | Senior Hiring Manager | filter out repeated applications | I do not have to potentially go through applicants who have been previously considered |
-| `* `  | Clumsy Hiring Manager | get a confirmation message when deleting an applicant | prevent accidental deletions of applicants |
-
-*{More to be added}*
+<div markdown="span" class="alert alert-info" role="alert">:information_source: <strong>Note:</strong> <br>
+The following legend describes the symbols used in this section:
+
+| Priority | Symbol  | Meaning          |
```
Comment (Merrickneo) [inline docs/DeveloperGuide.md:278]: Thanks for formatting the tables properly.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -324,19 +346,27 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+      Use case resumes at step 3.
+
+
+* 3b. The given applicant's phone number does not exist.
 
     * 3a1. HMHero shows an error message.
 
-      Use case resumes at step 2.
+      Use case resumes at step 3.
```
Comment (Merrickneo) [inline docs/DeveloperGuide.md:362]: Do we need to have a 3c for the case where the name and phone does not match?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -324,19 +346,27 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
 **Use case: Advance an applicant’s status**
 
 **MSS**
 
 1.  User requests to list applicants
 2.  HMHero shows a list of applicants
-3.  User requests to advance the status of a specific applicant in the list
+3.  User enters the command to advance the status of a specific applicant in the list
```
Comment (Merrickneo) [inline docs/DeveloperGuide.md:369]: Good edit to not specify the implementation details for 'HMHero'

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -384,66 +438,124 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    1. Applied (default, when applicant is added into HMHero)
     2. Shortlisted
     3. Accepted
     4. Rejected
-* **Application stage**:
+* **Application cycle**:
     * All applicants added are at the Applied status by default. From there, hiring managers can advance their
     application status to Shortlisted, then to Accepted. Applicants can be rejected at any stage excepted for Accepted.
-![Application Stage](images/application_stage.png)
+![Application Cycle](images/application_stage.png)
```
Comment (Merrickneo) [inline docs/DeveloperGuide.md:549]: I will send you the edited diagram! Looks a little off by overlaying the text over the box

**Review by Merrickneo — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -257,32 +257,48 @@ _{Explain here how the data archiving feature will be implemented}_
 
 **Target user profile**:
 
-* has a need to manage a significant number of applicants
-* prefer desktop apps over other types
-* can type fast
-* prefers typing to mouse interactions
-* is reasonably comfortable using CLI apps
+* Has a need to manage a significant number of applicants who applied to their department.
```
Comment (bofeng1999) [inline docs/DeveloperGuide.md:260]: Maybe can remove full stop, minor edit!

**Review by bofeng1999 — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (FredericChow00) [reply docs/DeveloperGuide.md:368]: edited!

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (FredericChow00) [reply docs/DeveloperGuide.md]: edited!

**Review by Merrickneo — APPROVED** [review]

**Pull request merged**